sdks/go/test/integration/primitives/timers_test.go

```diff
@@ -0,0 +1,41 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package primitives
+
+import (
+	"testing"
+	"time"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/ptest"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/transforms/periodic"
+	"github.com/apache/beam/sdks/v2/go/test/integration"
+)
+
+func TestTimers_EventTime_Bounded(t *testing.T) {
+	integration.CheckFilters(t)
+	ptest.BuildAndRun(t, TimersEventTime(beam.Impulse))
+}
+
+func TestTimers_EventTime_Unbounded(t *testing.T) {
+	integration.CheckFilters(t)
+	ptest.BuildAndRun(t, TimersEventTime(func(s beam.Scope) beam.PCollection {
+		now := time.Now()
+		return periodic.Impulse(s, now, now.Add(10*time.Second), 0, false)
+	}))
+}
```

Comment:
In the future might we anticipate, `TestTimers_ProcessTime_Bounded` and `TestTimers_ProcessTime_Unbounded` tests? As of this writing, for Java at least, `TestTimers_ProcessTime_Bounded` equivalent tests would fail.

Reply:
That is the intention yes. I agree that "bounded processing time" pipelines are likely to be invalid by construction. That would be Working as intended.

But I don't think I can write a ProcessingTime test without TestStream being available anyway, so it's moot, since such a test would be "unbounded" by default.

Reply:
Agreed. Please feel free to resolve this conversation.